### PR TITLE
Add MSG_STATUS_JOURNAL [4.1.x]

### PR DESCRIPTION
**Pull request by silverjam**
# Description

@swift-nav/devinfra

Adds new MSG_STATUS_JOURNAL, see https://swift-nav.atlassian.net/l/c/yYraQXHA

# API compatibility

_Does this change introduce a API compatibility risk?_ -> No, just introducing a new message.

# Related PRs
- https://github.com/swift-nav/libsbp/pull/1123

### CODE REVIEW

File: spec/yaml/swiftnav/sbp/system.yaml [hunk 1]
```diff
@@ -216,6 +216,85 @@ definitions:
...
+              - 0-15:
+                 desc: System
+                 values:
+                     - 0: Starling
+                     - 1: Precision GNSS Module (PGM)
+        - sbp_version:
+            type: u16
+            desc: SBP protocol version
+            fields:
+              - 8-16:
```
Comment (isaactorz) [inline spec/yaml/swiftnav/sbp/system.yaml:279]: Off by 1 error. Thanks Message Validator!

**Review by isaactorz — CHANGES_REQUESTED** [review]

File: spec/yaml/swiftnav/sbp/system.yaml (hunk 1 above)
Comment (silverjam) [reply spec/yaml/swiftnav/sbp/system.yaml]: Should be fixed now

**Review by isaactorz — APPROVED** [review]

File: rust/sbp/src/sbp_iter_ext.rs [hunk 2]
```diff
@@ -214,6 +214,7 @@ mod tests {
 
     use super::*;
 
+    #[cfg(feature = "swiftnav")]
```
Comment (silverjam) [inline rust/sbp/src/sbp_iter_ext.rs:217]: Changes to make VSCode happy (cc @notoriaga)

File: generator/sbpg/targets/resources/c/test/v4/sbp_c_test.c.j2 [hunk 3]
```diff
@@ -183,7 +183,7 @@ START_TEST( test_(((s.suite_name))) )
     sbp_message_send(&sbp_state, (((t.msg_type_name|convert_upper))), (((t.raw_json_obj.sender))), &test_msg, &dummy_write);
 
     ck_assert_msg(dummy_wr == sizeof(encoded_frame),
-        "not enough data was written to dummy_buff");
+        "not enough data was written to dummy_buff (expected: %zu, actual: %zu)", sizeof(encoded_frame), dummy_wr);
```
Comment (silverjam) [inline generator/sbpg/targets/resources/c/test/v4/sbp_c_test.c.j2:186]: Added to help with debugging

File: python/tests/sbp/utils.py [hunk 4]
```diff
@@ -94,7 +94,8 @@ def _assert_sbp(sbp, test_case):
     Unit test case parsed from YAML.
 
   """
-  assert sbp.crc == int(test_case['crc'], 0), "Invalid crc."
+  expected = int(test_case['crc'], 0)
+  assert sbp.crc == expected, "Invalid crc. Actual: {:x} vs expected: {:x}.".format(sbp.crc, expected)
```
Comment (silverjam) [inline python/tests/sbp/utils.py:98]: Debugging aid/improvement

**Review by notoriaga — APPROVED** [review]

**Review by RReichert — APPROVED** [review]
LGTM, thanks @silverjam .

This PR is currently being used by:

* https://github.com/swift-nav/swiftlets/pull/2057
* https://github.com/swift-nav/starling/pull/6514

**Review by woodfell — APPROVED** [review]

**Pull request merged**